src/frontend/js/components/CourseProductCourseRuns/index.spec.tsx

```diff
@@ -0,0 +1,372 @@
+      // a call to action to enroll.
+      expect(screen.getAllByRole('listitem')).toHaveLength(3);
+
+      // For each course run, it should render course, enrollment dates and a checkbox input
+      // to select the course run
+      courseRuns.forEach((courseRun: CourseRun) => {
+        // - Course run start date should be displayed
+        screen.getByText(dateFormatter.format(new Date(courseRun.start)), {
+          selector: '.course-runs-item__date.course-runs-item__date--start',
+        });
+
+        // - Course run end date should be displayed
+        screen.getByText(dateFormatter.format(new Date(courseRun.end)), {
+          selector: '.course-runs-item__date.course-runs-item__date--end',
+        });
+
+        // - Course run enrollment dates should be displayed
+        screen.getByText(
+          `Enrollment from ${dateFormatter.format(
+            new Date(courseRun.enrollment_start),
+          )} to ${dateFormatter.format(new Date(courseRun.enrollment_end))}`,
+          {
+            selector: '.course-runs-item__enrollment-dates',
+          },
+        );
+
+        // - A radio input
+        screen.getByRole('radio', {
+          name: `Select course run from ${dateFormatter.format(
+            new Date(courseRun.start),
+          )} to ${dateFormatter.format(new Date(courseRun.end))}.`,
+        });
+      });
+
+      // A call to action should be displayed ...
+      // ... when no course run is selected, it should be disabled
+      let $button: HTMLButtonElement = screen.getByRole('button', {
+        name: 'Select a course run',
+      });
+      expect($button.disabled).toBe(true);
+
+      // ... once a course run is selected, it should be enabled and allows user to enroll
+      await act(async () => {
+        // - Select the first course run
+        const $radio = screen.getByRole('radio', {
+          name: `Select course run from ${dateFormatter.format(
+            new Date(courseRuns[0].start),
+          )} to ${dateFormatter.format(new Date(courseRuns[0].end))}.`,
+        });
+        fireEvent.click($radio);
+      });
+
+      $button = screen.getByRole('button', {
+        name: 'Enroll',
+      });
+      expect($button.disabled).toBe(false);
+
+      // - User clicks to enroll
+      fetchMock.resetHistory();
+      const enrollmentDeferred = new Deferred();
+      fetchMock.post('https://joanie.test/api/enrollments/', enrollmentDeferred.promise);
+
+      await act(async () => {
+        fireEvent.click($button);
+      });
+
+      // A spinner should be displayed
+      screen.getByRole('status', { name: 'Enrolling...' });
+
+      await act(async () => {
+        enrollmentDeferred.resolve(200);
+      });
+
+      const calls = fetchMock.calls();
+      expect(calls).toHaveLength(2);
+      // A request to create the enrollment should have been executed
+      expect(calls[0][0]).toBe('https://joanie.test/api/enrollments/');
```

Comment:
```suggestion
      expect(calls[0][0]).toBe('https://joanie.test/api/enrollments/');
```
I think we should check the post are as expected, true for differents calls in your test files